src/main/java/com/narmnevis/range/io/PublisherCollection.java

```diff
@@ -31,6 +32,7 @@ public void publish(RangeContext context) {
 				publisher.publish(context);
 				logger.info("Publisher {} published context {}", publisher, context);
 			} catch (Exception e) {
+				logger.error("Exception occurred: ", ExceptionUtils.getRootCause(e));
```

Comment:
Refer to my comment on Guava, please. The same can be done in Guava by `Throwables` class.